systemtest/src/test/java/io/strimzi/systemtest/RollingUpdateST.java

```diff
@@ -269,6 +266,73 @@ void testKafkaAndZookeeperScaleUpScaleDown() throws Exception {
+        String topicName = "test-topic-" + new Random().nextInt(Integer.MAX_VALUE);
+        timeMeasuringSystem.setOperationID(timeMeasuringSystem.startTimeMeasuring(Operation.CLUSTER_RECOVERY));
+
+        KafkaResource.kafkaPersistent(CLUSTER_NAME, 3)
+            .editSpec()
+                .editKafka()
+                    .addToConfig("auto.create.topics.enable", "false")
+                .endKafka()
+            .endSpec().done();
+
+        LOGGER.info("Running kafkaScaleUpScaleDown {}", CLUSTER_NAME);
+        final int initialReplicas = kubeClient().getStatefulSet(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)).getStatus().getReplicas();
+        assertEquals(3, initialReplicas);
+
+        // scale up
+        final int scaleTo = initialReplicas + 4;
+        final int newPodId = initialReplicas;
+        final String newPodName = KafkaResources.kafkaPodName(CLUSTER_NAME,  newPodId);
+        LOGGER.info("Scaling up to {}", scaleTo);
+        // Create snapshot of current cluster
+        String kafkaStsName = kafkaStatefulSetName(CLUSTER_NAME);
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> k.getSpec().getKafka().setReplicas(scaleTo));
+        // No need to roll kafka cluster during scale up (no external listeners in Kafka)
+        StatefulSetUtils.waitForAllStatefulSetPodsReady(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), scaleTo);
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(kafkaStsName);
+        LOGGER.info("Scaling to {} finished", scaleTo);
+
+        // Create topic before scale up to ensure no partitions created on last broker (which will mess up scale down)
+        KafkaTopicResource.topic(CLUSTER_NAME, topicName, 4, 4, 4).done();
+
+        //Test that the new pod does not have errors or failures in events
+        String uid = kubeClient().getPodUid(newPodName);
+        List<Event> events = kubeClient().listEvents(uid);
+        assertThat(events, hasAllOfReasons(Scheduled, Pulled, Created, Started));
+
+        //Test that CO doesn't have any exceptions in log
+        timeMeasuringSystem.stopOperation(timeMeasuringSystem.getOperationID());
+        assertNoCoErrorsLogged(timeMeasuringSystem.getDurationInSecconds(testClass, testName, timeMeasuringSystem.getOperationID()));
+
+        // scale down
+        LOGGER.info("Scaling down to {}", initialReplicas);
+        timeMeasuringSystem.setOperationID(timeMeasuringSystem.startTimeMeasuring(Operation.SCALE_DOWN));
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> k.getSpec().getKafka().setReplicas(initialReplicas));
+
+        PodUtils.waitUntilPodsCountIsPresent(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), 3);
+        // Wait for first reconciliation
+        StUtils.waitForReconciliation(testClass, testName, NAMESPACE);
+        // Wait for second reconciliation and check that pods are not rollable
+        StUtils.waitForReconciliation(testClass, testName, NAMESPACE);
```

Comment:
What is expected to triger the rolling update here? And how do you check the 6.?

Reply:
After offline discussion added code for trigger rolling update manually and created issue for unexpected rolling update in that case.